source/includes/_tipser-react-elements.md

```diff
@@ -0,0 +1,263 @@
+ 
+#### `TipserElement` 
+Generic Element that can render any Contentful content that's fed as a prop to the element.
+
+#### `TipserProduct` 
+Element that renders the product given the Tipser product ID as the prop.
+
+#### `CartIcon` 
+Element that displays the number of items in your cart and gives the user a way to open the checkout dialog.
+
+
+
+### All configuration options of Tipser Elements
+
+Configuration is an object, which should be inserted into `TipserElementsProvider` as `config` prop.
+
+All properties are optional:
+
+property name | description |type | default value
+--------------|-------------|-----|---------------
+`defaultAddedToCartPopup` | controls whether default Added To Cart Popup is displayed | boolean | true 
+`eventsHandlers` | gives possibility to add handlers to Tipser Elements events | object | {}
+`useDefaultErrorHandler` | when set to false and error happens, default message won't be displayed | boolean | true
+
+### Event Handlers
+```js
+
+let tipserConfig = {
+    lang: 'en',
+    primaryColor: 'red',
+    // ----- EVENT HANDLING START
+    useDefaultErrorHandler: true,
+    eventsHandlers: {
+        onError: (error) => {
+            console.log(error)
+        },
+        onAddToCart: ({cartSize, product}) => {
+            console.log('Hurray, you have added item to cart. ', product);
+            console.log('Your cart size is now. ', cartSize);
+        }
+    }
+    // -- EVENT HANDLING END
+};
+
+// ... as in example above class App extends Component { ... }
+
+```
+Event handlers may be passed as a part of `config` option of `TipserElementsProvider`. There is a number of events exposed for developer. 
+
+
+Handler name  | description | params
+--------------|-------------|--------
+onError | Main goal of this handler is to add additional behavior when the error appears. | object of type EventError (see [EventError](#event-error-interface)
+onAddToCart | when product is being added to cart event is triggered | object of type {cartSize, product} where cartSize is a current size of cart after adding to cart and product is a product object with properties see (see [TipserProductModel](#tipser-product-model-interface)) 
+
+### Event Error interface
+
+```js 
+export interface EventError {
+    type?: string;
+    id?: string;
+    message?: string;
+    stack?: string;
+}
+```
+Param of `onError` handlers
+
+<br><br><br><br>
+
+### Tipser Product Model interface
+
+```js 
+export interface TipserProductModel {
+    id: string;
+    title: string;
+    description: string;
+    brand: string;
+    images: any[];
+    isInStock: boolean;
+    deliveryTime: string;
+    priceIncVat: PriceModel;
+    deliveryCost: PriceModel;
+    variants: TipserProductModel[];
+    discountPriceIncVat: PriceModel;
+    freeReturn: boolean;
+}
+```
+When onAddToCart is being dispatched the handlers are triggered with object with cart size of type string and product of type TipserProductModel.
```

Comment:
I think it's really hard to read that sentence and understand it without re-reading.

Reply:
yes, super complicated :(